imap_processing/codice/codice_l1a.py

```diff
@@ -115,12 +121,15 @@ def create_science_dataset(self, start_time: np.datetime64) -> xr.Dataset:
             np.arange(self.num_energy_steps),
             name="energy",
             dims=["energy"],
-            attrs=cdf_attrs.energy_attrs.output(),
+            attrs=cdf_attrs.variable_attributes["energy_attrs"],
```

Comment:
Similar to my comment to Sean, the schema validation step is run when the attributes are retrieved - so they should be accessed using `get_variable_attributes` instead. Alternatively, a property would also solve this, and eliminate a common source of confusion - so feel free to update `get_variable_attributes` to be a python property instead.